lib/providers/quoters/RoutingApiQuoter.ts

```diff
@@ -109,6 +156,14 @@ export class RoutingApiQuoter implements Quoter {
         ...(config.enableFeeOnTransferFeeFetching !== undefined && {
           enableFeeOnTransferFeeFetching: config.enableFeeOnTransferFeeFetching,
         }),
+        ...(request.info.type === TradeType.EXACT_OUTPUT && fullPortionPayloadForExactOut?.portionResponse?.hasPortion && {
+          portionAmount: fullPortionPayloadForExactOut.portionAmount?.quotient.toString() ?? undefined,
+          portionRecipient: fullPortionPayloadForExactOut.portionResponse.portion?.receiver,
+        }),
+        ...(request.info.type === TradeType.EXACT_INPUT && getPortionForExactIn?.hasPortion && {
```

Comment:
yea i guess we discussed this already but if the type was 
```
{
   portion: Portion | null;
}
```
instead of using `hasPortion`, you wouldn't have to add `?` on every line here, but can just check that `getPortionForExactIn.portion` in this conditional chain

Reply:
And you can't really use `!` below because its technically possible for `hasPortion = true` and `portion` to be undefined, which is impossible when you only have the one field `portion`.